### PR TITLE
Moving a bunch of integration test files to pytest.

### PR DESCRIPTION
**Pull request by prady0t**
Related to #4156  
Moving integration tests to pytest. These are the files I changed : 

`integration/test_models/test_full_battery_models/test_lithium_ion/test_spm_half_cell.py` `integration/test_models/test_full_battery_models/test_lithium_ion/test_basic_models.py` `integration/test_models/test_full_battery_models/test_lithium_ion/test_spme.py` `integration/test_models/test_full_battery_models/test_lithium_ion/test_dfn_half_cell.py` `integration/test_models/test_full_battery_models/test_lithium_ion/test_spme_half_cell.py` `integration/test_models/test_full_battery_models/test_lithium_ion/test_compare_basic_models.py` `integration/test_models/test_full_battery_models/test_lithium_ion/test_initial_soc.py` `integration/test_models/test_full_battery_models/test_lithium_ion/test_spm.py` `integration/test_models/test_full_battery_models/test_lithium_ion/test_yang2017.py` `integration/test_models/test_full_battery_models/test_lithium_ion/test_newman_tobias.py` `integration/test_models/test_full_battery_models/test_lithium_ion/test_external/test_external_temperature.py` `integration/test_models/test_equivalent_circuit/test_thevenin.py` `integration/test_models/test_lead_acid/test_compare_outputs.py` `integration/test_models/test_lead_acid/test_loqs.py` `integration/test_models/test_lead_acid/test_compare_basic_models.py` `integration/test_models/test_lead_acid/test_loqs_surface_form.py` `integration/test_models/test_lead_acid/test_asymptotics_convergence.py` `integration/test_models/test_lead_acid/test_full.py` `integration/test_models/test_lead_acid/test_side_reactions/test_loqs_side_reactions.py` `integration/test_models/test_lead_acid/test_side_reactions/test_full_side_reactions.py`
 
A total of 20 files (there are a total of 40 integration test files). These were the ones with less than 100 lines of code and fairly simple migrations.

### CODE REVIEW

File: tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_spme.py
Comment (agriyakhetarpal) [inline tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_spme.py:12]: Question: why do we need the fixtures here (and in other places)? Is this because `run_basic_processing_test` requires a model to be set up beforehand?

**Review by agriyakhetarpal — COMMENTED** [review]
Thanks, @prady0t, this looks good! This is removing the `TestCase` from a lot of files, I'll approve this after we merge #4180

**Review by agriyakhetarpal — APPROVED** [review]
Never mind, we have to remove the `TestCase` class from the test classes anyway. Approving this one with one question: https://github.com/pybamm-team/PyBaMM/pull/4238/files#r1664190217, we can merge after that is addressed.

File: tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_spme.py
Comment (prady0t) [reply tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_spme.py:12]: Yes and everywhere else as well. Actually unittest identifies `setUp` as test setup function but for pytest we have to mark them explicitly so that they are available everywhere.

Comment (agriyakhetarpal) [reply tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_spme.py:12]: I see – thanks for letting me know, resolving this conversation then!

**Comment by prady0t** [comment]
Why are tests failing?

**Comment by agriyakhetarpal** [comment]
CMake had a new 3.30 release which broke the cached `share/` directories for the installation. I'll manually delete some of the older caches and re-trigger the run.

**Comment by agriyakhetarpal** [comment]
Merging – let's hope nothing breaks. Please update the tracking issue with these files and the PR number.

**Pull request merged**
